internal/interface/interface.go

```diff
@@ -0,0 +1,46 @@
+package iface
```

Comment:
nit: Should the directory/filename also match this package name now that it's shortened?

The boilerplate should also go before any declarations.